cgen: missing method generation when calling method by generic method with multiple types

### Describe the bug

Issue was reported by thunder031275 on discord, made a smaller example which you can see in the repro.

Curiously this works if the parameter `obj T` is not `mut`.

### Reproduction Steps

```v
struct Aa {
}

fn (aa Aa) dump() {
	println(aa)
}

struct Bb {
}

fn (bb Bb) dump() {
	println(bb)
}

struct Foo {
}

fn (foo Foo) set[T](mut obj T) {
	obj.dump()
}

fn main() {
	f := Foo{}
	f.set(mut Aa{})
	f.set(mut Bb{})
}
```

### Expected Behavior

Should print out empty structs Aa and Bb.

### Current Behavior

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/dump.01K56S1F0A07M8KWYB0FKG22Y8.tmp.c:5243: warning: implicit declaration of function 'main__Aa_dump'
cc: /home/dylan/Repos/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
cc: tcc: error: undefined symbol 'main__Aa_dump'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

Generated user-defined C methods:
```c
VV_LOC void main__Bb_dump(main__Bb bb) {
	builtin__println(main__Bb_str(bb));
}
VV_LOC void main__Foo_set_T_main__Aa(main__Foo foo, main__Aa* obj) {
	main__Aa_dump(*obj);
}
VV_LOC void main__Foo_set_T_main__Bb(main__Foo foo, main__Bb* obj) {
	main__Bb_dump(*obj);
}
VV_LOC void main__main(void) {
	main__Foo f = ((main__Foo){E_STRUCT});
	main__Foo_set_T_main__Aa(f, (voidptr)&((main__Aa){E_STRUCT}));
	main__Foo_set_T_main__Bb(f, (voidptr)&((main__Bb){E_STRUCT}));
}
```
Notice that the declaration of `main__Aa_dump` is missing.

### V version

V 0.4.11 38ebb3f

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.11 b8260625f6358d9678d06c4ce8600a322ab2a717.38ebb3f
|:-------------------|:-------------------
|OS                  |linux, "CachyOS"
|Processor           |16 cpus, 64bit, little endian, AMD Ryzen 7 5700G with Radeon Graphics
|Memory              |7.42GB/13.55GB
|                    |
|V executable        |/home/dylan/Repos/v/v
|V last modified time|2025-09-15 00:26:01
|                    |
|V home dir          |OK, value: /home/dylan/Repos/v
|VMODULES            |OK, value: /home/dylan/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/dylan/Repos/v
|                    |
|Git version         |git version 2.51.0
|V git status        |weekly.2025.37-35-g38ebb3fb
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.2.1 20250813
|gcc version         |gcc (GCC) 15.2.1 20250813
|clang version       |clang version 20.1.8
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.42

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -35,6 +35,7 @@ mut:
 	all_decltypes map[string]ast.TypeDecl
 	all_structs   map[string]ast.StructDecl
 
+	cur_fn                 string
 	level                  int
 	is_builtin_mod         bool
 	is_direct_array_access bool
@@ -934,6 +935,7 @@ pub fn (mut w Walker) fn_decl(mut node ast.FnDecl) {
 	}
 	w.mark_fn_ret_and_params(node.return_type, node.params)
 	w.mark_fn_as_used(fkey)
+	w.cur_fn = fkey
 	w.stmts(node.stmts)
 	w.defer_stmts(node.defer_stmts)
 }
@@ -988,6 +990,19 @@ pub fn (mut w Walker) call_expr(mut node ast.CallExpr) {
 			if embed_types.len != 0 {
 				fn_embed := '${int(embed_types.last())}.${node.name}'
 				w.fn_by_name(fn_embed)
+			}
+		} else if node.left_type.has_flag(.generic) {
+			if w.cur_fn != '' {
+				if concrete_types_list := w.table.fn_generic_types[w.cur_fn] {
+					for concrete_types in concrete_types_list {
+						for concrete_type in concrete_types {
+							method_name := '${int(concrete_type)}.${node.name}'
+							if !w.used_fns[method_name] {
+								w.fn_by_name(method_name)
+							}
+						}
+					}
+				}
 			}
 		} else {
 			match left_sym.info {

--- a/vlib/v/tests/generics/generic_fn_multi_mutable_struct_test.v
+++ b/vlib/v/tests/generics/generic_fn_multi_mutable_struct_test.v
@@ -1,0 +1,26 @@
+struct Aa {
+}
+
+fn (aa Aa) dump() {
+	println(aa)
+}
+
+struct Bb {
+}
+
+fn (bb Bb) dump() {
+	println(bb)
+}
+
+struct Foo {
+}
+
+fn (foo Foo) set[T](mut obj T) {
+	obj.dump()
+}
+
+fn test_main() {
+	f := Foo{}
+	f.set(mut Aa{})
+	f.set(mut Bb{})
+}

